internal/handlers/irc/handlers.go

```diff
@@ -147,6 +148,25 @@ func kickHandler(c ClientInterface) func(*girc.Client, girc.Event) {
 	}
 }
 
+func nickHandler(c ClientInterface) func(*girc.Client, girc.Event) {
+	return func(gc *girc.Client, e girc.Event) {
+		c.Logger().LogDebug("nickHandler triggered")
+		if c.TgSettings().ShowNickMessage {
+			// e.Source.Name is the original name.
+			// e.Params[0] is the new nick name.
+			// However, let's assume it is possible (though unlikely)
+			// e.Params can be empty.
+			var newName string
```

Comment:
You could also do `newName := "Unspecified Name"` and save a tiny bit of typing, but this works.